### PR TITLE
Fix display of login buttons.

### LINKED ISSUE #1623: Add spacing between login buttons
screenshot from 2016-05-06 11 34 19

### PR DESCRIPTION
**Pull request by huwshimi**
Update the layout and display when there are multiple login buttons. Fixes #1623.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- update https://github.com/juju/juju-gui/blob/develop/jujugui/static/gui/src/app/app.js#L731 to return true
- in your browser console run `app.logout()`
- you should see two nicely laid out login buttons

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5010/
Test PASSed.

File: jujugui/static/gui/src/app/components/login/login.js [hunk 1]
```diff
@@ -98,6 +97,17 @@ YUI.add('login-component', function() {
     },
 
     render: function() {
+      var buttons = [];
+      var MacaroonButton = this._generateLoginWithMacaroonButton();
+      if (MacaroonButton) {
+        buttons.push(MacaroonButton);
+      }
+      buttons.push({
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/login/login.js]: I think this button should be the first one. When a user tabs from username to password then the next tab will be to login not login with SSO.

**Comment by anthonydillon** [comment]
Looks good with a comment in code about the order of the buttons.

I also get `Warning: Failed propType: Invalid prop `loginWithMacaroon` of type `boolean` supplied to `<<anonymous>>`, expected` when viewing the login screen. Not sure if that was introduced by this branch.

**Comment by kadams54** [comment]
Ditto on what @anthonydillon ran into in the console. I'd also like to get that error squashed if possible.

**Comment by huwshimi** [comment]
@anthonydillon @kadams54 that error appears for you because of the hack in the first QA step.

**Comment by huwshimi** [comment]
I've swapped the button order as suggested by @anthonydillon.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5024/
Test PASSed.

**Comment by anthonydillon** [comment]
Thanks :+1:

**Comment by kadams54** [comment]
:+1: for me as well, thanks!

**Comment by kadams54** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**